components/external-services-mock/internal/notification/handler.go

```diff
@@ -281,66 +232,28 @@ func (h *Handler) GetResponses(writer http.ResponseWriter, r *http.Request) {
 
 func (h *Handler) FailOnceResponse(writer http.ResponseWriter, r *http.Request) {
 	if h.ShouldReturnError {
-		id, ok := mux.Vars(r)["tenantId"]
-		if !ok {
-			httphelpers.WriteError(writer, errors.New("missing tenantId in url"), http.StatusBadRequest)
-			return
-		}
-
-		if _, ok = h.Mappings[id]; !ok {
-			h.Mappings[id] = make([]Response, 0, 1)
-		}
-		bodyBytes, err := ioutil.ReadAll(r.Body)
-		if err != nil {
-			httphelpers.WriteError(writer, errors.Wrap(err, "error while reading request body"), http.StatusInternalServerError)
-			return
-		}
-
-		var result interface{}
-		if err := json.Unmarshal(bodyBytes, &result); err != nil {
-			httphelpers.WriteError(writer, errors.Wrap(err, "body is not a valid JSON"), http.StatusBadRequest)
-			return
-		}
-
-		mappings := h.Mappings[id]
-		if r.Method == http.MethodPatch {
-			mappings = append(h.Mappings[id], Response{
-				Operation:   Assign,
-				RequestBody: bodyBytes,
-			})
-		}
-
-		if r.Method == http.MethodDelete {
-			applicationId, ok := mux.Vars(r)["applicationId"]
-			if !ok {
-				httphelpers.WriteError(writer, errors.New("missing applicationId in url"), http.StatusBadRequest)
-				return
+		responseFunc := func([]byte) {
+			response := struct {
+				Error string `json:"error"`
+			}{
+				Error: "failed to parse request",
```

Comment:
Couldn't we reuse the FAResponseBody with Error field in it and json tag `json:"error,omitempty"` and reuse it here?